crates/cli-support/src/js/mod.rs

```diff
@@ -478,6 +484,10 @@ impl<'a> Context<'a> {
         self.rewrite_imports(module_name);
         self.update_producers_section();
 
+        // Cause any future calls to `should_write_global` to panic, making sure
+        // we don't ask for items which we can no longer emit.
+        self.globals_written = true;
```

Comment:
Alternatively, `exposed_globals` could be an `Option<...>` and then we can have `should_write_global` do `self.exposed_globals.as_mut().unwrap().insert(name)` and then here do `self.exposed_globals.take().unwrap()`.

This is equivalent to having a bool flag (which becomes the option discriminant in this scheme) but it is *impossible* to forget to check the flag in `should_write_global` or anything else that wants to touch `exposed_globals` thanks to the type system.

Reply:
👍